<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to snifter • snifter</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to snifter">
<meta property="og:description" content="snifter">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">snifter</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/snifter.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Alanocallaghan/snifter/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="snifter_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to snifter</h1>
                        <h4 class="author">Alan O’Callaghan</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:alan.ocallaghan@outlook.com" class="email">alan.ocallaghan@outlook.com</a>
      
                  
      
      <small class="dont-index">Source: <a href="https://github.com/Alanocallaghan/snifter/blob/master/vignettes/snifter.Rmd"><code>vignettes/snifter.Rmd</code></a></small>
      <div class="hidden name"><code>snifter.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>snifter provides an R wrapper for the <a href="https://opentsne.readthedocs.io/en/latest/">openTSNE</a> implementation of fast interpolated t-SNE (FI-tSNE). It is based on <em><a href="https://bioconductor.org/packages/3.12/basilisk">basilisk</a></em> and <em><a href="https://CRAN.R-project.org/package=reticulate">reticulate</a></em>. This vignette aims to provide a brief overview of typical use when applied to scRNAseq data, but it does not provide a comprehensive guide to the available options in the package.</p>
<p>It is highly advisable to review the documentation in snifter and the <a href="https://opentsne.readthedocs.io/en/latest/">openTSNE documentation</a> to gain a full understanding of the available options.</p>
</div>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>We will illustrate the use of snifter using data from <em><a href="https://bioconductor.org/packages/3.12/scRNAseq">scRNAseq</a></em> and single cell utility functions provided by <em><a href="https://bioconductor.org/packages/3.12/scuttle">scuttle</a></em>, <em><a href="https://bioconductor.org/packages/3.12/scater">scater</a></em> and <em><a href="https://bioconductor.org/packages/3.12/scran">scran</a></em> - first we load these libraries and set a random seed to ensure the t-SNE visualisation is reproducible (note: it is good practice to ensure that a t-SNE embedding is robust by running the algorithm multiple times).</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"snifter"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"scRNAseq"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"scran"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"scuttle"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://bioconductor.org/packages/scater/">"scater"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></pre></div>
<p>Before running t-SNE, we first load data generated by <a href="https://doi.org/10.1126/science.aaa1934">Zeisel et al.</a> from <em><a href="https://bioconductor.org/packages/3.12/scRNAseq">scRNAseq</a></em>. We filter this data to remove genes expressed only in a small number of cells, estimate normalisation factors using <em><a href="https://bioconductor.org/packages/3.12/scran">scran</a></em> and generate 20 principal components. We will use these principal components to generate the t-SNE embedding later.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/ZeiselBrainData.html">ZeiselBrainData</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, <span class="op">]</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">data</span>, cluster <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickCluster.html">quickCluster</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">data</span>, ncomponents <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="co">## Convert this to a factor to use as colouring variable later</span>
<span class="va">data</span><span class="op">$</span><span class="va">level1class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span></pre></div>
</div>
<div id="running-t-sne" class="section level1">
<h1 class="hasAnchor">
<a href="#running-t-sne" class="anchor"></a>Running t-SNE</h1>
<p>The main functionality of the package lies in the <code>fitsne</code> function. This function returns a matrix of t-SNE co-ordinates. In this case, we pass in the 20 principal components computed based on the log-normalised counts. We colour points based on the discrete cell types identified by the authors.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu">reducedDim</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snifter.html">fitsne</a></span><span class="op">(</span><span class="va">mat</span>, random_state <span class="op">=</span> <span class="fl">42L</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">fit</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, colour <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>pch <span class="op">=</span> <span class="fl">19</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html">scale_colour_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Cell type"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"t-SNE 1"</span>, y <span class="op">=</span> <span class="st">"t-SNE 2"</span><span class="op">)</span></pre></div>
<p><img src="snifter_files/figure-html/run-1.png" width="700"></p>
</div>
<div id="projecting-new-data-into-an-existing-embedding" class="section level1">
<h1 class="hasAnchor">
<a href="#projecting-new-data-into-an-existing-embedding" class="anchor"></a>Projecting new data into an existing embedding</h1>
<p>The openTNSE package, and by extension snifter, also allows the embedding of new data into an existing t-SNE embedding. Here, we will split the data into “training” and “test” sets. Following this, we generate a t-SNE embedding using the training data, and project the test data into this embedding.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">test_ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">train_ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span><span class="op">)</span>, <span class="va">test_ind</span><span class="op">)</span>
<span class="va">train_mat</span> <span class="op">&lt;-</span> <span class="va">mat</span><span class="op">[</span><span class="va">train_ind</span>, <span class="op">]</span>
<span class="va">test_mat</span> <span class="op">&lt;-</span> <span class="va">mat</span><span class="op">[</span><span class="va">test_ind</span>, <span class="op">]</span>

<span class="va">train_label</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">level1class</span><span class="op">[</span><span class="va">train_ind</span><span class="op">]</span>
<span class="va">test_label</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">level1class</span><span class="op">[</span><span class="va">test_ind</span><span class="op">]</span>

<span class="va">embedding</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snifter.html">fitsne</a></span><span class="op">(</span><span class="va">train_mat</span>, random_state <span class="op">=</span> <span class="fl">42L</span><span class="op">)</span></pre></div>
<p>Once we have generated the embedding, we can now <code>project</code> the unseen test data into this t-SNE embedding.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">new_coords</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/project.html">project</a></span><span class="op">(</span><span class="va">embedding</span>, new <span class="op">=</span> <span class="va">test_mat</span>, old <span class="op">=</span> <span class="va">train_mat</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">embedding</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">embedding</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>,
            colour <span class="op">=</span> <span class="va">train_label</span>,
            shape <span class="op">=</span> <span class="st">"Train"</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">new_coords</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">new_coords</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, 
            colour <span class="op">=</span> <span class="va">test_label</span>,
            shape <span class="op">=</span> <span class="st">"Test"</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html">scale_colour_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Cell type"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_shape.html">scale_shape_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"t-SNE 1"</span>, y <span class="op">=</span> <span class="st">"t-SNE 2"</span><span class="op">)</span></pre></div>
<p><img src="snifter_files/figure-html/plot-embed-1.png" width="700"></p>
</div>
<div id="session-information" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.2 (2020-06-22)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel stats    graphics utils    stats4   methods  base    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scater_1.17.5               ggplot2_3.3.2              </span>
<span class="co">#&gt;  [3] scuttle_0.99.15             scran_1.17.18              </span>
<span class="co">#&gt;  [5] scRNAseq_2.3.13             SingleCellExperiment_1.11.7</span>
<span class="co">#&gt;  [7] SummarizedExperiment_1.19.8 Biobase_2.49.1             </span>
<span class="co">#&gt;  [9] GenomicRanges_1.41.6        GenomeInfoDb_1.25.11       </span>
<span class="co">#&gt; [11] IRanges_2.23.10             S4Vectors_0.27.13          </span>
<span class="co">#&gt; [13] BiocGenerics_0.35.4         MatrixGenerics_1.1.3       </span>
<span class="co">#&gt; [15] matrixStats_0.57.0          snifter_0.99.2             </span>
<span class="co">#&gt; [17] BiocStyle_2.17.1           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] ggbeeswarm_0.6.0              colorspace_1.4-1             </span>
<span class="co">#&gt;   [3] ellipsis_0.3.1                rprojroot_1.3-2              </span>
<span class="co">#&gt;   [5] bluster_0.99.1                XVector_0.29.3               </span>
<span class="co">#&gt;   [7] BiocNeighbors_1.7.0           fs_1.5.0                     </span>
<span class="co">#&gt;   [9] farver_2.0.3                  bit64_4.0.5                  </span>
<span class="co">#&gt;  [11] interactiveDisplayBase_1.27.5 AnnotationDbi_1.51.3         </span>
<span class="co">#&gt;  [13] knitr_1.30                    jsonlite_1.7.1               </span>
<span class="co">#&gt;  [15] dbplyr_1.4.4                  shiny_1.5.0                  </span>
<span class="co">#&gt;  [17] BiocManager_1.30.10           compiler_4.0.2               </span>
<span class="co">#&gt;  [19] httr_1.4.2                    dqrng_0.2.1                  </span>
<span class="co">#&gt;  [21] basilisk_1.1.17               backports_1.1.10             </span>
<span class="co">#&gt;  [23] assertthat_0.2.1              Matrix_1.2-18                </span>
<span class="co">#&gt;  [25] fastmap_1.0.1                 limma_3.45.14                </span>
<span class="co">#&gt;  [27] later_1.1.0.1                 BiocSingular_1.5.1           </span>
<span class="co">#&gt;  [29] htmltools_0.5.0               tools_4.0.2                  </span>
<span class="co">#&gt;  [31] rsvd_1.0.3                    igraph_1.2.5                 </span>
<span class="co">#&gt;  [33] gtable_0.3.0                  glue_1.4.2                   </span>
<span class="co">#&gt;  [35] GenomeInfoDbData_1.2.3        dplyr_1.0.2                  </span>
<span class="co">#&gt;  [37] grDevices_4.0.2               rappdirs_0.3.1               </span>
<span class="co">#&gt;  [39] Rcpp_1.0.5                    pkgdown_1.6.1                </span>
<span class="co">#&gt;  [41] vctrs_0.3.4                   ExperimentHub_1.15.3         </span>
<span class="co">#&gt;  [43] DelayedMatrixStats_1.11.1     xfun_0.18                    </span>
<span class="co">#&gt;  [45] stringr_1.4.0                 beachmat_2.5.7               </span>
<span class="co">#&gt;  [47] mime_0.9                      lifecycle_0.2.0              </span>
<span class="co">#&gt;  [49] irlba_2.3.3                   statmod_1.4.34               </span>
<span class="co">#&gt;  [51] AnnotationHub_2.21.5          edgeR_3.31.4                 </span>
<span class="co">#&gt;  [53] zlibbioc_1.35.0               scales_1.1.1                 </span>
<span class="co">#&gt;  [55] basilisk.utils_1.1.11         ragg_0.3.1                   </span>
<span class="co">#&gt;  [57] promises_1.1.1                yaml_2.2.1                   </span>
<span class="co">#&gt;  [59] curl_4.3                      gridExtra_2.3                </span>
<span class="co">#&gt;  [61] memoise_1.1.0                 reticulate_1.16              </span>
<span class="co">#&gt;  [63] datasets_4.0.2                stringi_1.5.3                </span>
<span class="co">#&gt;  [65] RSQLite_2.2.1                 BiocVersion_3.12.0           </span>
<span class="co">#&gt;  [67] desc_1.2.0                    filelock_1.0.2               </span>
<span class="co">#&gt;  [69] BiocParallel_1.23.2           rlang_0.4.7                  </span>
<span class="co">#&gt;  [71] pkgconfig_2.0.3               systemfonts_0.3.2            </span>
<span class="co">#&gt;  [73] bitops_1.0-6                  evaluate_0.14                </span>
<span class="co">#&gt;  [75] lattice_0.20-41               purrr_0.3.4                  </span>
<span class="co">#&gt;  [77] labeling_0.3                  bit_4.0.4                    </span>
<span class="co">#&gt;  [79] tidyselect_1.1.0              magrittr_1.5                 </span>
<span class="co">#&gt;  [81] bookdown_0.20                 R6_2.4.1                     </span>
<span class="co">#&gt;  [83] generics_0.0.2                DelayedArray_0.15.12         </span>
<span class="co">#&gt;  [85] DBI_1.1.0                     pillar_1.4.6                 </span>
<span class="co">#&gt;  [87] withr_2.3.0                   RCurl_1.98-1.2               </span>
<span class="co">#&gt;  [89] tibble_3.0.3                  crayon_1.3.4                 </span>
<span class="co">#&gt;  [91] BiocFileCache_1.13.1          rmarkdown_2.4                </span>
<span class="co">#&gt;  [93] viridis_0.5.1                 locfit_1.5-9.4               </span>
<span class="co">#&gt;  [95] grid_4.0.2                    blob_1.2.1                   </span>
<span class="co">#&gt;  [97] digest_0.6.25                 xtable_1.8-4                 </span>
<span class="co">#&gt;  [99] httpuv_1.5.4                  munsell_0.5.0                </span>
<span class="co">#&gt; [101] viridisLite_0.3.0             beeswarm_0.2.3               </span>
<span class="co">#&gt; [103] vipor_0.4.5</span></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alan O'Callaghan, Aaron Lun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
